themes/openy_themes/openy_rose/templates/node/node--branch--teaser.html.twig

```diff
@@ -81,13 +81,12 @@
   ]
 %}
 <article{{ attributes.addClass(classes) }}>
-
   {{ content.field_location_area }}
 
   {{ title_prefix }}
   {% if not page %}
     <h2{{ title_attributes }}>
-      <a href="{{ url }}" rel="bookmark">{{ label }}</a>
+      <a href="{{ node.field_location_temp_url.0.url ? node.field_location_temp_url.0.url : url }}" rel="bookmark">{{ label }}</a>
```

Comment:
1. This should be applied only when branch is "Coming Soon".
2. Let's set variable using `set` and the re-use it. `node.field_location_temp_url.0.url` looks a little  bit ugly.